src/engine/SCons/Util.py

```diff
@@ -1175,44 +1175,12 @@ def unique(s):
     return u
 
 
-
-# From Alex Martelli,
-# http://aspn.activestate.com/ASPN/Cookbook/Python/Recipe/52560
-# ASPN: Python Cookbook: Remove duplicates from a sequence
-# First comment, dated 2001/10/13.
-# (Also in the printed Python Cookbook.)
-# This not currently used, in favor of the next function...
-
-def uniquer(seq, idfun=None):
-    def default_idfun(x):
-        return x
-    if not idfun:
-        idfun = default_idfun
-    seen = {}
-    result = []
-    for item in seq:
-        marker = idfun(item)
-        # in old Python versions:
-        # if seen.has_key(marker)
-        # but in new ones:
-        if marker in seen: continue
-        seen[marker] = 1
-        result.append(item)
-    return result
-
 # A more efficient implementation of Alex's uniquer(), this avoids the
 # idfun() argument and function-call overhead by assuming that all
 # items in the sequence are hashable.
 
 def uniquer_hashables(seq):
-    seen = {}
-    result = []
-    for item in seq:
-        #if not item in seen:
-        if item not in seen:
-            seen[item] = 1
-            result.append(item)
-    return result
+    return list(set(seq))
```

Comment:
How about this: http://code.activestate.com/recipes/52560/#c3

Reply:
Things have evolved, there aren't a lot of 17-year old Python recipes that are still the best answer...